### PR TITLE
ELEMENTS-1327: unlock polymer version

### PR DESCRIPTION
**Pull request by richardsd**
In short, for the data table the selection problem comes from two code paths where **this.$.list.toggleSelectionForIndex** is called:
- [onCheckboxTap handler](https://github.com/nuxeo/nuxeo-elements/blob/4df9e4d26c473da55b9fbc4f5f3c80b4375e8cad/ui/nuxeo-data-table/iron-data-table.js#L852)
- [_selectionHandler](https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/nuxeo-data-table/iron-data-table.js#L573) that is registered as listener in the [page-provider-display-behavior](https://github.com/nuxeo/nuxeo-elements/blob/4df9e4d26c473da55b9fbc4f5f3c80b4375e8cad/ui/nuxeo-page-provider-display-behavior.js#L413) (when `selectOnTap` is set to false)

### CODE REVIEW

File: ui/nuxeo-data-table/iron-data-table.js [hunk 1]
```diff
@@ -849,7 +849,7 @@ import '../nuxeo-button-styles.js';
 
     _onCheckBoxTap(e) {
       if (this.selectionEnabled && !this.selectAllActive) {
-        this.$.list.toggleSelectionForIndex(e.model.index);
+        // this.$.list.toggleSelectionForIndex(e.model.index);
```
Comment (mnixo) [inline ui/nuxeo-data-table/iron-data-table.js]: Should we add more information about why we're commenting this line out?

**Review by mnixo — DISMISSED** [review]

File: ui/nuxeo-data-table/iron-data-table.js (hunk 1 above)
Comment (richardsd) [reply ui/nuxeo-data-table/iron-data-table.js]: I think we should remove the line and maybe add a comment explaining the behavior. wdyt?

**Comment by richardsd** [comment]
Cross repo build https://github.com/nuxeo/nuxeo-web-ui/runs/4534055450 🔵

**Review by semisse — DISMISSED** [review]

File: ui/nuxeo-page-provider-display-behavior.js [hunk 2]
```diff
@@ -389,11 +389,21 @@ export const PageProviderDisplayBehavior = [
...
-      return !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(this.items[index]) > -1);
+      return this.multiSelection
+        ? !!(
+            this.selectedItems &&
+            this.selectedItems.length &&
+            this.items &&
+            this.items.length > index &&
+            this.selectedItems.indexOf(this.items[index]) > -1
+          )
+        : !!(this.selectedItem && this.items && this.items.length > index && this.selectedItem === this.items[index]);
```
Comment (Gabez0r) [inline ui/nuxeo-page-provider-display-behavior.js:406]: Should we add some unit tests for this?

**Review by Gabez0r — COMMENTED** [review]
LGTM.

File: ui/nuxeo-page-provider-display-behavior.js (hunk 2 above)
Comment (richardsd) [reply ui/nuxeo-page-provider-display-behavior.js:406]: Maybe it's a good idea 👍  Let me see how we can improve it.

**Comment by richardsd** [comment]
Added unit tests to cover the selection helpers update.

**Review by Gabez0r — APPROVED** [review]

**Review by mnixo — APPROVED** [review]

**Pull request merged**